iOS/Controllers/Lesson/SingleLessonViewController.swift

```diff
@@ -32,22 +38,22 @@ class SingleLessonViewController: UITableViewController {
         self.title = lesson.name
     }
 
-    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return self.contents.count
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.observer = NotificationCenter.default.addObserver(forName: UIContentSizeCategory.didChangeNotification,
+                                                               object: nil,
+                                                               queue: .main) { [unowned self] _ in
+            self.buildContentCache()
+        }
```

Comment:
Please move this code to `viewDidLoad`. Removing the observer is not required as of iOS 9. Then there is also no need for a member variable for the observer.

https://useyourloaf.com/blog/unregistering-nsnotificationcenter-observers-in-ios-9/
https://developer.apple.com/documentation/foundation/nsnotificationcenter/1415360-addobserver

Reply:
I'm using block based observing, and that needs to be removed from the notification center: https://developer.apple.com/documentation/foundation/notificationcenter/1411723-addobserver
I can use selector based API if you prefer.